lens/private/list/flatten.scrbl

```diff
@@ -0,0 +1,63 @@
+#lang scribble/manual
+
+@(require "../doc-util/main.rkt")
+
+@title{Flattening and unflattening lists}
+
+@defthing[append*-lens lens?]{
+A lens that flattens a list one-level down when viewing, and restores
+the original structure when setting.
+@lens-unstable-examples[
+  (lens-view append*-lens '((a) (b c) () (d e f)))
+  (lens-set append*-lens '((a) (b c) () (d e f)) '(1 2 3 4 5 6))
+]}
+
+@defproc[(flatten/depth-lens [n exact-nonnegative-integer?]) lens?]{
+Creates a lens that flattens a list of depth @racket[n] when
+viewing, and restores the original structure when setting.
+
+Conceptually, @racket[append*-lens] is equivalent to
+@racket[(flatten/depth-lens 2)], since it takes a list of depth 2.
```

Comment:
`append*-lens` should also describe this relationship so they're both reachable from each other.